Improve the README (#515)
meilisearch/milli: README.md

This repository contains crates to quickly debug the engine:
 - There are benchmarks located in the `benchmarks` crate.
 - The `cli` crate is a simple command-line interface that helps run [flamegraph] on top of it.
 - The `filter-parser` crate contains the parser for the Meilisearch filter syntax.

>  - The `flatten-serde-json` crate contains the library that flattens serde-json `Value` objects like elastic search does.

Reviewer: ```suggestion
 - The `flatten-serde-json` crate contains the library that flattens serde-json `Value` objects like Elasticsearch does.
```

 - The `helpers` crate is only used to do operations on the database.

>  - The `http-ui` crate is a simple HTTP dashboard to tests the features like for real!

Reviewer: ```suggestion
 - The `http-ui` crate is a simple HTTP dashboard to test the features like for real!
```

 - The `infos` crate is used to dump the internal data-structure and ensure correctness.
 - The `json-depth-checker` crate is used to indicate if a JSON must be flattened.

## How to use it?

Milli is a library that does search things, it must be embedded in a program.
You can compute the documentation of it by using `cargo doc --open`.

Here is an example usage of the library where we insert documents into the engine

> and search for one of them just after.

Reviewer: ```suggestion
and search for one of them right after.
```
